controller/openid_configuration.go

```diff
@@ -0,0 +1,86 @@
+package controller
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/client"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/goadesign/goa"
+)
+
+// OpenidConfigurationController implements the openid_configuration resource.
+type OpenidConfigurationController struct {
+	*goa.Controller
+	configuration LoginConfiguration
+}
+
+// NewOpenidConfigurationController creates a openid_configuration controller.
+func NewOpenidConfigurationController(service *goa.Service, configuration LoginConfiguration) *OpenidConfigurationController {
+	return &OpenidConfigurationController{Controller: service.NewController("OpenidConfigurationController"), configuration: configuration}
+}
+
+// Show runs the show action.
+func (c *OpenidConfigurationController) Show(ctx *app.ShowOpenidConfigurationContext) error {
+
+	// Getting Keycloak's /.well-known/openid-configuration
+	keycloakOpenIDConfigurationEndpoint := c.configuration.GetKeycloakURL() + "/auth/realms/" + c.configuration.GetKeycloakRealm() + "/.well-known/openid-configuration"
+	response, err := http.Get(keycloakOpenIDConfigurationEndpoint)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{}, "request to achieve openid-configuration of keycloak failed")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		panic(err.Error())
+	}
+	keycloakOpenIDConfiguration := &app.OpenIDConfiguration{}
+	err = json.Unmarshal(body, keycloakOpenIDConfiguration)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{}, "unable to unmashal to json")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	issuer := rest.AbsoluteURL(ctx.RequestData, "")
+	authorizationEndpoint := rest.AbsoluteURL(ctx.RequestData, client.AuthorizeAuthorizePath())
+	tokenEndpoint := rest.AbsoluteURL(ctx.RequestData, client.ExchangeTokenPath())
+	logoutEndpoint := rest.AbsoluteURL(ctx.RequestData, client.LogoutLogoutPath())
+	jwksURI := c.configuration.GetKeycloakEndpointCerts()
+
+	authOpenIDConfiguration := &app.OpenIDConfiguration{
+		// REQUIRED properties
+		Issuer:                 &issuer,
+		AuthorizationEndpoint:  &authorizationEndpoint,
+		TokenEndpoint:          &tokenEndpoint,
+		EndSessionEndpoint:     &logoutEndpoint,
+		ResponseTypesSupported: []string{"code"},
+		JwksURI:                &jwksURI,
```

Comment:
Use auth.osio/api/token/keys endpoint URL instead of Keycloak URL.

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/277/commits/2af8a55888dcbefdd6b9a45c4064855c9c56ff32